ansible_collections/arista/avd/plugins/plugin_utils/cv_client/workflows/deploy_cv_pathfinder_metadata_to_cv.py

```diff
@@ -286,38 +343,52 @@ async def deploy_cv_pathfinder_metadata_to_cv(cv_pathfinder_metadata: list[CVPat
         update_general_metadata(metadata=pathfinders[0].metadata, studio_inputs=studio_inputs)
 
     for pathfinder in pathfinders:
-        upsert_pathfinder(metadata=pathfinder.metadata, device=pathfinder.device, studio_inputs=studio_inputs)
+        result.warnings.extend(
+            upsert_pathfinder(metadata=pathfinder.metadata, device=pathfinder.device, studio_inputs=studio_inputs, studio_version=studio_version)
+        )
 
     for edge in edges:
-        upsert_edge(metadata=edge.metadata, device=edge.device, studio_inputs=studio_inputs)
+        result.warnings.extend(upsert_edge(metadata=edge.metadata, device=edge.device, studio_inputs=studio_inputs, studio_version=studio_version))
 
     if studio_inputs != existing_studio_inputs:
         await cv_client.set_studio_inputs(studio_id=CV_PATHFINDER_METADATA_STUDIO_ID, workspace_id=result.workspace.id, inputs=studio_inputs)
 
     result.deployed_cv_pathfinder_metadata.extend(pathfinders + edges)
 
 
-def generate_internet_exit_metadata(metadata: dict, device: CVDevice) -> dict:
+def generate_internet_exit_metadata(metadata: dict, device: CVDevice, studio_version: StudioVersion) -> tuple[dict, list[str]]:
     """
     Generate internet-exit related metadata for one device.
     To be inserted into edge router metadata under "services"
+
+    Returns metadata dict and list of any warnings raised.
     """
     if (internet_exit_policies := get(metadata, "internet_exit_policies")) is None:
-        LOGGER.info("deploy_cv_pathfinder_metadata_to_cv: Did not find 'internet_exit_policies' for device: %s", device.hostname)
-        return []
+        LOGGER.debug("deploy_cv_pathfinder_metadata_to_cv: Did not find 'internet_exit_policies' for device: %s", device.hostname)
+        return {}, []
 
     LOGGER.info("deploy_cv_pathfinder_metadata_to_cv: Found %s 'internet_exit_policies' for device: %s", len(internet_exit_policies), device.hostname)
 
     services_dict = {}
+    warnings = []
+
     for internet_exit_policy in internet_exit_policies:
         # We currently only support zscaler
-        if internet_exit_policy.get("type") != "zscaler":
-            LOGGER.info(
-                "deploy_cv_pathfinder_metadata_to_cv: Ignoring unsupported internet exit policy '%s' with type '%s' for device: %s.",
-                internet_exit_policies.get("name"),
-                internet_exit_policy.get("type"),
-                device.hostname,
+        if internet_exit_policy.get("type") not in ["zscaler", "direct"]:
+            warning = (
+                f"deploy_cv_pathfinder_metadata_to_cv: Ignoring unsupported internet exit policy '{internet_exit_policies.get('name')}' "
+                f"with type '{internet_exit_policy.get('type')}' for device: {device.hostname}."
+            )
+            LOGGER.info(warning)
```

Comment:
Should the warning be added to `warnings` here as well just like in line 391?

Reply:
No this is only for the debug output.

Reply:
so in one case you have `warnings.append(warning)` for a non supported feature and in the other case you don't - it does not feel consistent

Reply:
I did not want to make noise for "direct" or other future types, but now I added handling of "direct, so I can make it as a warning now.